judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/ContestRoleChecker.java

```diff
@@ -48,8 +55,10 @@ public ContestRole getRole(String userJid, Contest contest) {
             return ContestRole.MANAGER;
         } else if (canSupervise(userJid, contest)) {
             return ContestRole.SUPERVISOR;
+        } else if (contestContestantDao.selectByContestJidAndUserJid(contest.getJid(), userJid).isPresent()) {
```

Comment:
You can just use `ContestRoleDao::isContestant()` here

(so `ContestContestantDao` is not needed)